src/notification.c

```diff
@@ -305,11 +305,11 @@ int notification_init(notification * n, int id)
         n->urls = notification_extract_markup_urls(&(n->body));
 
         n->msg = string_replace_all("\\n", "\n", g_strdup(n->format));
-        n->msg = notification_replace_format("%a", n->appname, n->msg,
+        n->msg = notification_replace_format("%a", string_cut_maxlen(n->appname, MAX_CHARS), n->msg,
```

Comment:
Oops, I missed this earlier but this creates a memory leak since the memory you allocate in `string_cut_maxlen` is never freed. Additionally I'm not sure if having a limit per-field is the correct option. In my opinion I'd change this to a single `string_cut_maxlen` call at the end of the format replacements.

Also, it should probably be placed after the icon processing, since we have flags to add the icon path to the format and a huge icon path would cause the same problem.